python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -859,18 +896,40 @@ def _log_asset_materialization_events_for_asset(
             f"Unexpected asset execution type {execution_type}",
         )
 
-        yield from (
-            (
-                DagsterEvent.asset_materialization(step_context, materialization)
-                for materialization in _get_output_asset_materializations(
-                    asset_key,
-                    partitions,
-                    output,
-                    output_def,
-                    manager_metadata,
-                    step_context,
+        # This is a temporary workaround to prevent duplicate observation events from external
+        # observable assets that were auto-converted from source assets. These assets yield
+        # observation events through the context in their body, and will continue to do so until we
```

Comment:
Is this referring to the code in `wrap_source_asset_observe_fn_in_op_compute_fn`?

Reply:
yes, I've clarified the comment